### DISCUSSION TITLE
Persistant to Disk in Valkey

### DISCUSSION THREAD

**Discussion by robertpcontreras-ts** [valkey-io/valkey#1221] [Q&A] [answered]
Appreciate this might be quite a unique use case but we use Redis atm as the driver for all of our queue'd jobs.

A problem we run into every now and again is the instance filling up with too many jobs and exhausting the available memory.

This leads to new jobs not being able to be pushed to the queue and some data loss.

Is there any way (or plans) with Valkey to start persisting to a disk in this this scnenario so that the instance continues working as expected.

This is different to snapshots as this only stores data in memory on disk in case of restart.

Appreciate this will impact performance if required, but its better than the instance just not accepting new records.

**Comment by PingXie** [comment, marked as answer]
I think your question is related to the data tiering discussion at https://github.com/valkey-io/valkey/issues/83. Please feel free to add your thoughts there.

**Reply by robertpcontreras-ts** [reply]
Thankyou!
